core/src/components/input-password-toggle/input-password-toggle.tsx

```diff
@@ -0,0 +1,141 @@
+import type { ComponentInterface } from '@stencil/core';
+import { Component, Element, Host, Prop, h, forceUpdate } from '@stencil/core';
+import { printIonWarning } from '@utils/logging';
+import { createColorClasses } from '@utils/theme';
+import { eyeOff, eye } from 'ionicons/icons';
+
+import { getIonMode } from '../../global/ionic-global';
+import type { Color } from '../../interface';
+
+/**
+ * @virtualProp {"ios" | "md"} mode - The mode determines which platform styles to use.
+ */
+@Component({
+  tag: 'ion-input-password-toggle',
+  styleUrls: {
+    ios: 'input-password-toggle.ios.scss',
+    md: 'input-password-toggle.md.scss',
+  },
+  shadow: true,
+})
+export class InputPasswordToggle implements ComponentInterface {
+  private inputElRef!: HTMLIonInputElement | null;
+
+  @Element() el!: HTMLIonInputElement;
+
+  /**
+   * The color to use from your application's color palette.
+   * Default options are: `"primary"`, `"secondary"`, `"tertiary"`, `"success"`, `"warning"`, `"danger"`, `"light"`, `"medium"`, and `"dark"`.
+   * For more information on colors, see [theming](/docs/theming/basics).
+   */
+  @Prop({ reflect: true }) color?: Color;
+
+  /**
+   * The icon that can be used to represent showing a password. If not set, the "eye" Ionicon will be used.
+   */
+  @Prop() showPasswordIcon?: string;
+
+  /**
+   * The icon that can be used to represent hiding a password. If not set, the "eyeOff" Ionicon will be used.
+   */
+  @Prop() hidePasswordIcon?: string;
```

Comment:
The term "password" here may be redundant. What do we think of doing `showIcon` or `hideIcon` (or something similar?).

Reply:
Agree it is redundant, I'd vote for `showIcon` and `hideIcon`.

Reply:
Renamed in https://github.com/ionic-team/ionic-framework/pull/29175/commits/90e62e394fd900147affb3627c8d102fcb8310d4